pandas/core/groupby.py

```diff
@@ -4437,20 +4437,15 @@ def _reorder_by_uniques(uniques, labels):
 
 
 def _groupby_indices(values):
+    """
+    return a dictionary of groups -> indices
+    """
+    if not is_categorical_dtype(values):
+        values = Categorical(values)
```

Comment:
Oh, I see it now, I went one class too deep.  I do think you could speed up the single grouping case  here - https://github.com/jreback/pandas/blob/580237924022eb74575420ad4433952c8de318dd/pandas/core/groupby.py#L2345 - make it:
```
return self.index.groupby(Categorical.from_codes(self.labels, self.group_index))
```